### PR TITLE
Logging: rethrowAsync facility

### PR DESCRIPTION
**Pull request by dvoytenko**
This API is necessary to preserve user/dev information that may already be present in the error.

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -332,6 +316,46 @@ function pushIfNonEmpty(array, val) {
...
+  } else if (message) {
+    error.message = message + ': ' + error.message;
+  }
+  return error;
+}
+
+
+/**
+ * Rethrows the error without terminating the current context. This preserves
+ * whether the original error designation as a user error or a dev error.
```
Comment (erwinmombay) [inline src/log.js]: is a

File: src/log.js [hunk 2]
```diff
@@ -211,12 +195,12 @@ export class Log {
...
-   * @param {string|!Error} errorOrMessage
+   * @param {...*} var_args
    * @return {!Error}
    */
-  createError(errorOrMessage) {
-    const error = typeof errorOrMessage == 'string' ?
-        new Error(errorOrMessage) : errorOrMessage;
+  createError(var_args) {
+    const error = createErrorVargs.apply(null,
+        Array.prototype.slice.call(arguments, 0));
```
Comment (erwinmombay) [inline src/log.js]: no need to slice here and just do `apply(null, arguments)`

File: test/functional/test-log.js [hunk 3]
```diff
@@ -445,4 +446,71 @@ describe('Logging', () => {
...
+    });
+
+    it('should rethrow error with single message', () => {
+      rethrowAsync('intended');
+      expect(() => {
+        clock.tick(1);
+      }).to.throw(Error, /^intended$/);
+    });
+
+    it('should rethrow error a single error', () => {
```
Comment (erwinmombay) [inline test/functional/test-log.js]: rethrow a single error?

**Comment by erwinmombay** [comment]
@dvoytenko LGTM,  + minor nits

File: src/log.js (hunk 1 above)
Comment (dvoytenko) [inline src/log.js]: Done

File: src/log.js (hunk 2 above)
Comment (dvoytenko) [inline src/log.js]: Done

File: test/functional/test-log.js (hunk 3 above)
Comment (dvoytenko) [inline test/functional/test-log.js]: Done

**Pull request merged**